conclave-host/src/main/kotlin/com/r3/conclave/host/internal/gramine/GramineEnclaveHandle.kt

```diff
@@ -150,20 +158,46 @@ class GramineEnclaveHandle(
         return false
     }
 
+    private fun unTarFile(tarGz: Path, outputDir: Path) {
+        tarGz.inputStream().use { fis ->
+            GZIPInputStream(fis).use { gis ->
+                TarArchiveInputStream(gis).use { tis ->
+                    var tarEntry = tis.nextTarEntry
+
+                    while (tarEntry != null) {
+                        val outputFile = outputDir / tarEntry.name
+                        if (tarEntry.isDirectory) {
+                            if (!outputFile.exists()) {
+                                outputFile.toFile().mkdirs()
+                            }
+                        } else {
+                            outputDir.toFile().mkdirs()
+                            FileOutputStream(outputFile.toFile()).use {
+                                IOUtils.copy(tis, it)
+                            }
```

Comment:
Use `Files.copy`

Reply:
I think that what I was trying to do here was to copy from the stream to the file, is `Files.copy` providing this?

Reply:
Yes